### ISSUE TITLE
Accessibility improvements

### ISSUE THREAD

**Issue by tylertadej** [Stanko/react-animate-height#96]
It seems like the `aria-hidden` attribute cannot be overwritten. It would be great if consumers could either overwrite `aria-hidden` to `undefined` or if this component would be updated with some accessibility improvements.

**Code example**

Example taken from: https://www.w3.org/WAI/GL/wiki/Using_the_WAI-ARIA_aria-expanded_state_to_mark_expandable_and_collapsible_regions

```
<p class="button">
    <button id="button1" class="buttonControl" aria-controls="t1" aria-expanded="false"><span>Show</span> Topic 1</button>
</p>

<div id="t1" class="topic" role="region" tabindex="-1" >
    Topic 1 is all about being Topic 1 and may or may not have anything to do with other topics.
</div>
```

**Additional context**
If you'd like a pull request, I can open one with either approach:

1. Allow `aria-hidden` to be overwritten.
2. Remove `aria-hidden` and add `aria-expanded` and `aria-controls` attributes.

Consumers should probably be able to control `role` and `tabIndex` but the accessibility attributes seem to make sense here given the simplicity and intended use of the component.

Thanks!

**Comment by Stanko** [comment]
Hello @tylertadej,

This is a valid concern, and I'll try to break it down so I can easier wrap my had around it.

Library already allows passing `role`, `tabIndex` or any `aria-` props:
https://github.com/Stanko/react-animate-height/blob/master/source/AnimateHeight.js#L352

As for the `aria-hidden`, I think you are completely right, it can probably be removed completely (content already has `display: none`) and replaced with `aria-expanded`.

So it would look something like this:

```js
    return (
      <div
        { ...omit(this.props, ...PROPS_TO_OMIT) }
        aria-expanded={ height === 0 }
      >
```

**Summary**

Library will automatically take care of `aria-expanded` automatically, but user can pass `aria-controls` (or any other `aria-` prop) manually. Content div will still be hidden by `display: none` when height is 0.

Let me know what do you think?

**Comment by tylertadej** [comment]
Thank you for the quick response!

> Library will automatically take care of `aria-expanded` automatically

Yes, and it seems that `aria-expanded` should be on the `button` element.

This resource has a good overview of the levels of configuration: https://www.accede-web.com/en/guidelines/rich-interface-components/show-hide-panels/

Bootstrap uses a similar configuration: https://getbootstrap.com/docs/4.0/components/collapse/

> user can pass `aria-controls` (or any other `aria-` prop) manually. Content div will still be hidden by `display: none` when height is 0.

Perfect.

**Comment by RavenAlly** [comment]
Agree with @tylertadej, `aria-expanded` should be applied to the `button` for maximum effectiveness rather than the `div`.
Agree with @Stanko 's approach to make `aria-controls` and other ARIA markup manual, the only mandatory item is `aria-expanded`.
Content hidden using `display:none` is perfect, it won't get passed through to assistive technology when it's collapsed (height 0). I like the approach of tying `aria-expanded` value to the height if that's how panel state is calculated.
Nice work!

**Comment by Stanko** [comment]
My bad for `aria-expandable` but I have to admit that I'm a little bit confused now.

This library wasn't intended to include a button. It is on the user to implement button in the way it fits the specific needs.

Isn't this what you are trying to achieve:
```jsx
render() {
  const { height } = this.state;

  return (
    <div>
      <button aria-expanded={height !== 0} aria-controls="example-panel">
        Toggle
      </button>
      <AnimateHeight id="example-panel">
        Content goes here
      </AnimateHeight>
    </div>
  )
}
```

Here you can see the live example:
https://codesandbox.io/s/react-animate-height-basic-example-8ktqy?file=/src/index.js

**Comment by tylertadej** [comment]
Apologies for the confusion. The sandbox is correct.

I think the improvements are:

- Consumers can configure `aria-hidden` as they choose.
- Library documentation includes a recommended accessibility implementation.

**Comment by Stanko** [comment]
That makes sense 👍 I'll make changes and publish it over the weekend.

**Comment by Stanko** [comment]
Published in `2.0.23` 🎉 
https://github.com/Stanko/react-animate-height/commit/d6e10162805da90cc1a0fde3163f7309cb822600

Let me know if anything else should be added.

**Comment by tylertadej** [comment]
Awesome. Thank you for the quick implementation!
